src/tree/ActionTreeItem.ts

```diff
@@ -0,0 +1,100 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IncomingMessage } from 'ms-rest';
+import * as path from 'path';
+import { gitHubWebResource } from 'vscode-azureappservice/out/src/github/connectToGitHub';
+import { requestUtils } from 'vscode-azureappservice/out/src/utils/requestUtils';
+import { AzureTreeItem, TreeItemIconPath } from "vscode-azureextensionui";
+import { IGitHubContext } from '../commands/github/IGitHubContext';
+import { ext } from '../extensionVariables';
+import { delay } from '../utils/delay';
+import { createGitHubRequestOptions } from '../utils/gitHubUtils';
+import { localize } from '../utils/localize';
+import { treeUtils } from "../utils/treeUtils";
+import { ActionsTreeItem } from "./ActionsTreeItem";
+
+export type GitHubAction = {
+    id: string; conclusion: 'success' | 'failure' | 'skip' | 'cancelled' | null;
+    event: string;
+    head_branch: string;
+    status: 'queued' | 'in-progress';
+    head_commit: { message: string };
+    url: string;
+    html_url: string;
+    rerun_url: string;
+    cancel_url: string;
+};
+
+export class ActionTreeItem extends AzureTreeItem {
+
+    public static contextValue: string = 'azureStaticAction';
+    public readonly contextValue: string = ActionTreeItem.contextValue;
+    public parent: ActionsTreeItem;
+    public data: GitHubAction;
+
+    constructor(parent: ActionsTreeItem, data: GitHubAction) {
+        super(parent);
+        this.data = data;
+    }
+
+    public get iconPath(): TreeItemIconPath {
+        return this.data.conclusion ? treeUtils.getThemedIconPath(path.join('conclusions', this.data.conclusion)) : treeUtils.getThemedIconPath(path.join('statuses', this.data.status));
+    }
+
+    public get id(): string {
+        return `${this.parent.parent.id}/${this.data.id}`;
```

Comment:
It's not needed, but just lets us know from the id which treeItem it's under. I think all the ids should be unique as is.

Reply:
It's just weird that it's the grandparent instead of the parent. That being said, `fullId` already includes the parent id, so there's no value in adding parent id to `id`